src/sensors.cpp

```diff
@@ -209,6 +216,28 @@ void Sensors::update_other_sensors()
       data_.sonar_range_valid = sonar_outlier_filt_.update(raw_distance, &data_.sonar_range);
     }
     break;
+  case BATTERY_MONITOR:
+      if(rf_.board_.clock_millis() - last_battery_monitor_update_ms > BATTERY_MONITOR_UPDATE_PERIOD)
+      {
+        last_battery_monitor_update_ms = rf_.board_.clock_millis();
+        if (rf_.board_.battery_voltage_present())
+        {
+          data_.battery_monitor_present = true;
+          battery_voltage_history[battery_montitor_filter_pointer++]=rf_.board_.battery_voltage_read();
+          if(battery_montitor_filter_pointer == BATTERY_MONITOR_MOVING_AVERAGE_COUNT)
+            battery_montitor_filter_pointer = 0;
+          float voltage_sum{0};
+          for(float voltage: battery_voltage_history)
+            voltage_sum += voltage;
+          data_.battery_voltage = voltage_sum / BATTERY_MONITOR_MOVING_AVERAGE_COUNT;
+        }
+        if(rf_.board_.battery_current_present())
+        {
+          data_.battery_monitor_present = true;
+          data_.battery_current = rf_.board_.battery_current_read();
```

Comment:
There is a lot of logic here.

Could we move this to another function?  (Is there a reason we can't use the outlier_filter?)

Reply:
I'm moving to another function. The purpose here is not to remove outliers, but to filter the pretty severe noise. The monitor should really have a hardware filter.